packages/components/datepicker/src/Datepicker.tsx

```diff
@@ -1,27 +1,177 @@
-import React from 'react';
+import React, {
+  ChangeEventHandler,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
 import { cx } from 'emotion';
 import { CommonProps } from '@contentful/f36-core';
 
+import { format, isValid, parse, startOfDay, endOfDay } from 'date-fns';
 import { getStyles } from './Datepicker.styles';
 import { Calendar } from './Calendar';
+import { Popover } from '@contentful/f36-popover';
+import { DayPickerSingleProps } from 'react-day-picker';
+import FocusLock from 'react-focus-lock';
+import { TextInput, TextInputProps } from '@contentful/f36-forms';
+import { IconButton } from '@contentful/f36-button';
+import { CalendarIcon } from '@contentful/f36-icons';
+import { PopoverProps } from '@contentful/f36-popover/src';
 
-export type DatepickerProps = CommonProps;
+const DATE_FORMAT = 'dd/MM/yyyy';
 
-function _Datepicker(props: DatepickerProps, ref: React.Ref<HTMLDivElement>) {
+export type DatepickerProps = CommonProps & {
+  onSelect: (day: Date | undefined) => void;
+} & Omit<
+    DayPickerSingleProps,
+    'mode' | 'onSelect' | 'fromMonth' | 'toMonth' | 'fromYear' | 'toYear'
+  > &
+  Pick<
+    TextInputProps,
+    'id' | 'isDisabled' | 'isInvalid' | 'isReadOnly' | 'isRequired'
+  > &
+  Pick<PopoverProps, 'usePortal'>;
+
+/**
+ * Provides functionality for date selection.
+ */
+export function Datepicker(props: DatepickerProps) {
   const styles = getStyles();
+  const {
+    //common props
+    testId = 'cf-datepicker',
+    className,
+    style,
+    // input props
```

Comment:
Do you think we should group these? 
```js
{
  //...
  inputProps: { id, isDisabled, ...otherInputProps },
  popoverProps: { ... },
  calendarProps: { ... },
  //...
}
```
Users might need to pass some more of the components props then they'll have to add it here first

Reply:
I don't like the idea of having these separate prop objects for each component we use inside.
For example props like `selected`, `onSelect`, `fromDate`, `toDate` - are not specifically calendar Props they are used to define input behavior as well. Same about input props like `isDisabled`, `isInvalid`, they are not about input only, but about the whole Datepicker.
If we want it to be such a flexible component we should might consider make it a compound one, or smth similar with render functions.
Otherwise, users should see it as one complete component and they don't care what it's built from, IMO.

What do you think? Maybe you can provide some references where such pattern is used well, the one with grouped props.

Reply:
I think we should split anything going directly to subcomponents. We used to have a style guide to define this pattern, but I can't find it at the moment. @mshaaban0 do you have it?

Reply:
Just to be clear, this can be in another PR once we agree on an approach